python/lsst/meas/extensions/shapeHSM/_hsm_shape.py

```diff
@@ -0,0 +1,340 @@
+    """Base plugin for HSM shape measurement."""
+
+    ConfigClass = HsmShapeConfig
+    doc = ""
+
+    def __init__(self, config, name, schema, metadata, logName=None):
+        if logName is None:
+            logName = __name__
+        super().__init__(config, name, schema, metadata, logName=logName)
+
+        # Define flags for possible issues that might arise during measurement.
+        flagDefs = measBase.FlagDefinitionList()
+        self.FAILURE = flagDefs.addFailureFlag("General failure flag, set if anything went wrong")
+        self.NO_PIXELS = flagDefs.add("flag_no_pixels", "No pixels to measure")
+        self.NOT_CONTAINED = flagDefs.add(
+            "flag_not_contained", "Center not contained in footprint bounding box"
+        )
+        self.PARENT_SOURCE = flagDefs.add("flag_parent_source", "Parent source, ignored")
+        self.GALSIM = flagDefs.add("flag_galsim", "GalSim failure")
+
+        # Embed the flag definitions in the schema using a flag handler.
+        self.flagHandler = measBase.FlagHandler.addFields(schema, name, flagDefs)
+
+        # Utilize a safe centroid extractor that uses the detection footprint
+        # as a fallback if necessary.
+        self.centroidExtractor = measBase.SafeCentroidExtractor(schema, name)
+
+        # Define symbol for measurement type based on shearType.
+        # "g": g-type shear, "e": e-type distortion.
+        self.measTypeSymbol = "g" if config.shearType == "KSB" else "e"
+
+        self.e1Key = self._addEllipticityField(name, 1, self.measTypeSymbol, schema, self.doc)
+        self.e2Key = self._addEllipticityField(name, 2, self.measTypeSymbol, schema, self.doc)
+        self.sigmaKey = schema.addField(schema.join(name, "sigma"), type=float, doc=f"{self.doc} (width)")
+        self.resolutionKey = schema.addField(
+            schema.join(name, "resolution"), type=float, doc="Resolution factor (0=unresolved, 1=resolved)"
+        )
+        self.hasDeblendKey = len(config.deblendNChild) > 0
+
+        if self.hasDeblendKey:
+            self.deblendKey = schema[config.deblendNChild]
+
+        self.log = logging.getLogger(self.logName)
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.SHAPE_ORDER
+
+    def _addEllipticityField(self, name, n, measTypeSymbol, schema, doc):
+        """
+        Helper function to add an ellipticity field to a measurement schema.
+
+        Parameters
+        ----------
+        name : str
+            Base name of the field.
+        n : int
+            Specifies whether the field is for the first (1) or second (2)
+            component.
+        measTypeSymbol : MeasType
+            The symbol for measurement type, either 'e' or 'g'.
+        schema : lsst.afw.table.Schema
+            The schema to which the field is added.
+        doc : str
+            The documentation string that needs to be updated to reflect the
+            type and component of the measurement.
```

Comment:
Since `measTypeSumbol` and `doc` can be fetched from `self`, these can be dropped from the method's signature.